animations/base_clip.py

```diff
@@ -0,0 +1,92 @@
+from moviepy.editor import *
+
+
+class BaseClip:
+
+    def __init__(self):
+        self.clip = None
+
+    def add_text(self, text, font_size, color, font, interline, pos, duration):
+        text = TextClip(text, fontsize=font_size, color=color,
+                        font=font, interline=interline).set_pos(pos).set_duration(duration)
+        self.clip = CompositeVideoClip([self.clip, text])
+
+    def resize(self, new_size):
+        """
+        Uses moviepy.video.fx.all.resize module
+        :param new_size: Can be wither(width,height) in pixels or a float
+                         A scaling factor, like 0.5
+                         A function of time returning one of these.
+        """
+        self.clip = self.clip.resize(new_size)
+
+    def crop(self, aspect_ratio=None, x1=None, y1=None, x2=None, y2=None,
```

Comment:
There may be a complication we should consider. Do we want to be able to crop the individual clips and the overall video, or just the overall video?

Reply:
It could be an option for the end user. Or we could enforce that cropping be only done on the overall video